packages/spawner/src/coveredCall.js

```diff
@@ -14,8 +14,47 @@ import { mustBeSameStructure, sameStructure } from '@agoric/same-structure';
  * limited, and the offerer and potential acceptor are 'bob' and 'alice'
  * respectively.
  */
-const coveredCall = harden({
+export const makeContract = harden({
   start: (terms, inviteMaker) => {
+
+    function checkUnits(installation, allegedInviteUnits, expectedTerms) {
+      mustBeSameStructure(
+        allegedInviteUnits.extent.installation,
+        installation,
+        'coveredCall checkUnits installation',
+      );
+      const [termsMoney, termsStock, termsTimer, termsDeadline] = expectedTerms;
+      const allegedInviteTerms = allegedInviteUnits.extent.terms;
+      const allegedInviteMoney = allegedInviteTerms.money;
+      if (allegedInviteMoney.extent !== termsMoney.extent) {
```

Comment:
Covered call is not specific to nat (fungible natural numbered) goods, so this is the wrong equality. You should use `!sameStructure(...)`

Reply:
Mark, I think you're focusing on the correctness of spawner, but I think this PR is about dynamic vat creation in general. I filed this particular problem as an issue a while back: https://github.com/Agoric/agoric-sdk/issues/86

Reply:
@Chris-Hibbert asked me to look at the contractHost contracts as translated to live under the new spawner. I see you closed #86 because we are shelving contractHost. Although this does lower its priority, we should still try to keep it correct if we can at reasonable cost.

If we close a bug without fixing it, please let's put a comment in the source text indicating that the code at that location is known to be incorrect.